### PR TITLE
Update dev dependencies

### PR DESCRIPTION
**Pull request by scepticulous**
Eslint
----

* Update eslint
* Update config packages
* Disabled new rules to avoid massive diff churn
* Apply import/order rule to all files
* Add peer dependency `acorn` (see below)

Added `acorn` as a dev dependency to meet `acorn-jsx` requirements:
`npm WARN acorn-jsx@5.0.1 requires a peer of acorn@^6.0.0 but none is installed. You must install peer dependencies yourself.`

Full hierarchy: 
```
/Users/me/dev/work/github/integrations/slack
└─┬ eslint@4.15.0
  └─┬ espree@3.5.2
    └── acorn-jsx@3.0.1
```

Others
---

* Update supertest

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,10 +39,11 @@
...
   },
   "devDependencies": {
-    "eslint": "^4.14.0",
-    "eslint-config-airbnb-base": "^12.1.0",
+    "eslint": "^6.1.0",
+    "eslint-config-airbnb-base": "^14.0.0",
     "eslint-config-probot": ">=0.1.0",
-    "eslint-plugin-import": "^2.13.0",
+    "eslint-plugin-import": "^2.18.2",
+    "acorn": "^6.0.0",
```
Comment (wilhelmklopp) [inline package.json]: Is `acorn` now required as a peer dep by one of our existing packages?

Comment (scepticulous) [reply package.json]: It is a peer dependency of one of our dependencies, yes.

Comment (wilhelmklopp) [reply package.json]: Do you know which one? I could probably find out if I went digging through that `package-lock.json` file 😇

Comment (scepticulous) [reply package.json]: Haha, sure. I only mentioned the direct parent in the PR body, which is `acorn-jsx`. The full chain is:
```
[...]/dev/work/github/integrations/slack
└─┬ eslint@4.15.0
  └─┬ espree@3.5.2
    └── acorn-jsx@3.0.1
```

**Pull request merged**